subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractLazyModuleComponentResolveMetadata.java

```diff
@@ -188,6 +189,23 @@ private void populateHierarchy(Configuration metadata, ImmutableSet.Builder<Stri
         }
     }
 
+    private ImmutableSet<String> constructVariantHierarchy(ConfigurationMetadata base, AdditionalVariant additionalVariant) {
```

Comment:
I do not understand the objective of this. Gradle variants have no hierarchy. Is it a left over from iterative attempts?
According to my understanding (and running the tests with the getter back to `ImmutableSet.of(name)`), this is not used by the other changes.
Otherwise, can you explain the reason this is needed?

Reply:
oh good call, this can be gone. Having the excludes carry over is just enough. Thanks